charmdir_test.go

```diff
@@ -100,6 +99,52 @@ func (s *CharmDirSuite) TestArchiveTo(c *gc.C) {
 	s.assertArchiveTo(c, baseDir, charmDir)
 }
 
+func (s *CharmDirSuite) TestArchiveToWithVersionString(c *gc.C) {
+	baseDir := c.MkDir()
+	charmDir := cloneDir(c, charmDirPath(c, "dummy"))
+
+	dir, err := charm.ReadCharmDir(charmDir)
+	c.Assert(err, gc.IsNil)
+
+	// create an empty .execName file inside tempDir
+	vcsPath := filepath.Join(dir.Path, ".git")
+	_, err = os.Create(vcsPath)
+	c.Assert(err, jc.ErrorIsNil)
+
+	path := filepath.Join(baseDir, "archive.charm")
+	file, err := os.Create(path)
+	c.Assert(err, gc.IsNil)
+
+	testing.PatchExecutableAsEchoArgs(c, s, "git")
+
+	err = dir.ArchiveTo(file)
+	file.Close()
+	c.Assert(err, gc.IsNil)
+
+	args := []string{"describe", "--dirty"}
+	testing.AssertEchoArgs(c, "git", args...)
+
+	zipr, err := zip.OpenReader(path)
+	c.Assert(err, gc.IsNil)
+	defer zipr.Close()
+
+	var verf *zip.File
+	for _, f := range zipr.File {
+		switch f.Name {
```

Comment:
this can just be an if statement, no need for a switch
and if there's a match, break from the loop

Reply:
done